src/Bridge/Laravel/Http/Middlewares/AuditMiddleware.php

```diff
@@ -0,0 +1,86 @@
+<?php
+declare(strict_types=1);
+
+namespace LoyaltyCorp\Auditing\Bridge\Laravel\Http\Middlewares;
+
+use Closure;
+use Illuminate\Http\Request;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Symfony\Bridge\PsrHttpMessage\HttpMessageFactoryInterface;
+use Symfony\Component\HttpFoundation\Response;
+
+class AuditMiddleware
+{
+    /**
+     * @var \Symfony\Bridge\PsrHttpMessage\HttpMessageFactoryInterface
+     */
+    private $psr7Factory;
+
+    /**
+     * AuditMiddleware constructor.
+     *
+     * @param \Symfony\Bridge\PsrHttpMessage\HttpMessageFactoryInterface $psr7Factory
+     */
+    public function __construct(HttpMessageFactoryInterface $psr7Factory)
+    {
+        $this->psr7Factory = $psr7Factory;
+    }
+
+    /**
+     * Handle the request and response and convert them to psr7 to be logged
+     *
+     * @param \Illuminate\Http\Request $request
+     * @param \Closure $next
+     *
+     * @return mixed
+     */
+    public function handle(Request $request, Closure $next)
+    {
+        try {
+            $response = $next($request);
+        } catch (\Exception $exception) {
+            // TODO: returning null until a decision is made on how to handle exceptions
```

Comment:
Please create a ticket for this TODO

Reply:
so we are going to re-throw the exception as I updated in the code, or do we still need discussion on this?

Reply:
Nope, rethrowing is good.